### PR TITLE
Add support for edition 2021.

### PR DESCRIPTION
**Pull request by m-ou-se**
This adds initial support for Rust 2021, by making it behave the same as Rust 2018. It simply changes `== Edition2018` to `>= Edition2018`. (And adds a `PartialOrd` implementation to `Edition` to make that possible.)

### CODE REVIEW

**Comment by calebcartwright** [comment]
Thanks for this, we've triggered an early publish of the rustc-ap* crates so v697 is now available, and we should be able to proceed a bit earlier than Tue

**Comment by m-ou-se** [comment]
Updated.

This now fails on a change in how rustfmt formats macro arguments:

```diff
-macro foo($type_name: ident, $docs: expr) {
+macro foo($type_name:ident, $docs:expr) {
```

Trying to find out which rustc change caused this.

**Comment by m-ou-se** [comment]
Looks like this is a bug that is now fixed by that change.

Before that change, `format_snippet` inside `MacroBranch::rewrite` would return `None` for that test case. After that change, it returns `Some(..)` with the newly formatted macro definition with the two spaces removed.

**Comment by m-ou-se** [comment]
Ah, the problem was that rustfmt skipped formatting of the entire macro definition because of `#[doc=$docs]` inside the body, which was solved by that change in `rustc_parse`. See https://github.com/rust-lang/rust/pull/78837#issuecomment-753400518

I'll update the expected test output, as this seems simply a bug that is now fixed.

**Review by calebcartwright — COMMENTED** [review]
Changes LGTM, thank you! Would you mind adding a test that has the Edition set to 2021? 

Something as simple as adding a single file (perhaps `tests/target/imports_2021_edition.rs`) with a snippet like the one below would suffice
```rust
// rustfmt-edition: 2021

use ::foo::Bar;
```

**Comment by m-ou-se** [comment]
Done!

File: tests/target/imports_2021_edition.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// rustfmt-edition: 2021
+
+use ::happy::new::year;
```
Comment (calebcartwright) [inline tests/target/imports_2021_edition.rs:3]: :laughing: love it

**Pull request merged**